### PR TITLE
fix: err create application when repoUrl with a trailing slash (#9857)

### PR DESCRIPTION
**Pull request by yyzxw**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Fixes [#9857]

### CODE REVIEW

**Comment by ashutosh16** [comment]
@yyzxw  Will, you be able to add a unit test for this change.

**Comment by yyzxw** [comment]
@ashutosh16 Thank you for review, I have added unit tests for this change, can you help me review it again?

File: util/argo/argo.go [hunk 1]
```diff
@@ -153,6 +153,10 @@ func ValidateRepo(
 	spec := &app.Spec
 	conditions := make([]argoappv1.ApplicationCondition, 0)
 
+	if strings.HasSuffix(spec.Source.RepoURL, "/") {
+		return nil, apierr.NewBadRequest(fmt.Sprintf("repository URL '%s' ends with '/', it seems to be consistent with '%s'", spec.Source.RepoURL, strings.TrimSuffix(spec.Source.RepoURL, "/")))
```
[the remark this answers is not in the document]
Comment (yyzxw) [reply util/argo/argo.go]: Oh, sorry, I've corrected it

**Comment by ashutosh16** [comment]
thanks, LGTM
@crenshaw-dev  Can you review the PR and merge?

**Review by ashutosh16 — APPROVED** [review]
LG

**Comment by yyzxw** [comment]
@ashutosh16 This PR has not been merged. Please help me merge it again, thank you

**Comment by ashutosh16** [comment]
@yyzxw  @crenshaw-dev wants to relook the changes before approving/merging.

**Review by blakepettersson — APPROVED** [review]

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
I don't think this is the right approach. It's acceptable to use a repo with a trailing slash. The problem is when you accidentally use a trailing slash when the configured repo does not have one.

See this comment for an explanation of how I think we ought to fix the issue: https://github.com/argoproj/argo-cd/issues/9857#issuecomment-1175049809

**Pull request closed without merging**